src/utils/durableUtils.ts

```diff
@@ -163,19 +155,28 @@ export namespace durableUtils {
 
     async function installDotnetDependencies(context: IFunctionWizardContext): Promise<void> {
         const packageNames: string[] = [];
+        const isDotnetIsolated: boolean = /Isolated/i.test(context.projectTemplateKey ?? '');
+
         switch (context.newDurableStorageType) {
             case DurableBackend.Netherite:
-                packageNames.push(durableUtils.dotnetDfNetheritePackage);
+                isDotnetIsolated ?
+                    packageNames.push(dotnetIsolatedDfNetheritePackage) :
+                    packageNames.push(dotnetLtsDfNetheritePackage);
                 break;
             case DurableBackend.SQL:
-                packageNames.push(durableUtils.dotnetDfSqlPackage);
+                isDotnetIsolated ?
+                    packageNames.push(dotnetIsolatedDfSqlPackage) :
+                    packageNames.push(dotnetLtsDfSqlPackage);
                 break;
             case DurableBackend.Storage:
             default:
         }
 
-        // Seems that the package arrives out-dated and needs to be updated
-        packageNames.push(durableUtils.dotnetDfBasePackage);
+        // Seems that the package arrives out-dated and needs to be updated to at least 2.9.1;
+        // otherwise, error appears when running with sql backend
+        if (!isDotnetIsolated) {
+            packageNames.push(dotnetLtsDfBasePackage);
```

Comment:
Not 100% sure if this is the best way to handle this; let me know if anyone has opinions on if this should be handled differently.